tests/components/essent/test_init.py

```diff
@@ -0,0 +1,153 @@
+"""Integration tests for Essent setup and runtime."""
+
+from __future__ import annotations
+
+import pytest
+from freezegun.api import FrozenDateTimeFactory
+from essent_dynamic_pricing import (
+    EssentConnectionError,
+    EssentDataError,
+    EssentError,
+    EssentResponseError,
+)
+
+from homeassistant.components.essent.const import DOMAIN, UPDATE_INTERVAL
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.util import dt as dt_util
+
+from tests.common import async_fire_time_changed
+
+from . import setup_integration
+
+pytestmark = [
+    pytest.mark.freeze_time("2025-11-16 10:30:00+01:00"),
+]
+
+
+async def test_full_integration_setup(hass: HomeAssistant) -> None:
+    """Test complete integration setup and unload."""
+    entry = await setup_integration(hass)
+    ent_reg = er.async_get(hass)
+
+    updated = False
+    for unique_id in ("electricity_next_price", "gas_next_price"):
+        entity_id = ent_reg.async_get_entity_id("sensor", "essent", unique_id)
+        assert entity_id is not None
+        reg_entry = ent_reg.async_get(entity_id)
+        assert reg_entry is not None
+        if reg_entry.disabled_by:
+            ent_reg.async_update_entity(entity_id, disabled_by=None)
+            updated = True
+
+    if updated:
+        await hass.config_entries.async_reload(entry.entry_id)
+        await hass.async_block_till_done()
+
+    assert entry.state == ConfigEntryState.LOADED
+
+    def _state(unique_id: str) -> str | None:
+        entity_id = ent_reg.async_get_entity_id("sensor", "essent", unique_id)
+        assert entity_id is not None
+        state = hass.states.get(entity_id)
+        assert state is not None
+        return state.state
+
+    assert _state("electricity_current_price") is not None
+    assert _state("electricity_next_price") is not None
+    assert _state("gas_current_price") is not None
+    assert _state("gas_next_price") is not None
+
+    assert await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+    assert entry.state == ConfigEntryState.NOT_LOADED
+
+
+async def test_device_registry_entry(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+) -> None:
+    """Ensure the integration registers a service device."""
+    entry = await setup_integration(hass)
+
+    device_entry = device_registry.async_get_device(
+        identifiers={(DOMAIN, entry.entry_id)}
+    )
+    assert device_entry is not None
+    assert device_entry.manufacturer == "Essent"
+    assert device_entry.name == "Essent"
+    assert device_entry.entry_type is dr.DeviceEntryType.SERVICE
```

Comment:
```suggestion
    device_registry: dr.DeviceRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Ensure the integration registers a service device."""
    entry = await setup_integration(hass)

    device_entry = device_registry.async_get_device(
        identifiers={(DOMAIN, entry.entry_id)}
    )
    assert device_entry is not None
    assert device_entry == snapshot
```
If you have this code, you can execute `pytest ./tests/components/essent --snapshot-update` and it will generate an .ambr file with the device